x/stakeibc/keeper/msg_server.go

```diff
@@ -1103,3 +1106,66 @@ func (k msgServer) ResumeHostZone(goCtx context.Context, msg *types.MsgResumeHos
 
 	return &types.MsgResumeHostZoneResponse{}, nil
 }
+
+// Submits an ICA tx to either grant or revoke authz permisssions to an address
+// to execute trades on behalf of the trade ICA
+func (k msgServer) ToggleTradeController(
+	goCtx context.Context,
+	msg *types.MsgToggleTradeController,
+) (*types.MsgToggleTradeControllerResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// Fetch the trade ICA which will be the granter
+	tradeRoute, found := k.GetTradeRouteFromTradeChainId(ctx, msg.ChainId)
+	if !found {
+		return nil, types.ErrTradeRouteNotFound.Wrapf("trade route not found for chain ID %s", msg.ChainId)
+	}
+
+	swapMsgTypeUrl := "/" + proto.MessageName(&types.SwapAmountInRoute{})
```

Comment:
Shouldn't this be MsgSwapExactAmountIn?  The SwapAmountInRoute is a type which goes into one of the fields of this message but the actual tx they would be running is the MsgSwapExactAmountIn

Reply:
great eye! you're 100% right - just updated